habitat/robots/static_manipulator.py

```diff
@@ -0,0 +1,488 @@
+        self.sim_obj.joint_positions = joint_positions
+
+    def set_gripper_target_state(self, gripper_state: float) -> None:
+        """Set the gripper motors to a desired symmetric state of the gripper [0,1] -> [open, closed]"""
+        for i, jidx in enumerate(self.params.gripper_joints):
+            delta = (
+                self.params.gripper_closed_state[i]
+                - self.params.gripper_open_state[i]
+            )
+            target = self.params.gripper_open_state[i] + delta * gripper_state
+            self._set_motor_pos(jidx, target)
+
+    def close_gripper(self) -> None:
+        """Set gripper to the close state"""
+        self.set_gripper_target_state(1)
+
+    def open_gripper(self) -> None:
+        """Set gripper to the open state"""
+        self.set_gripper_target_state(0)
+
+    @property
+    def is_gripper_open(self) -> bool:
+        """True if all gripper joints are within eps of the open state."""
+        return (
+            np.amax(
+                np.abs(
+                    self.gripper_joint_pos
+                    - np.array(self.params.gripper_open_state)
+                )
+            )
+            < self.params.gripper_state_eps
+        )
+
+    @property
+    def is_gripper_closed(self) -> bool:
+        """True if all gripper joints are within eps of the closed state."""
+        return (
+            np.amax(
+                np.abs(
+                    self.gripper_joint_pos
+                    - np.array(self.params.gripper_closed_state)
+                )
+            )
+            < self.params.gripper_state_eps
+        )
+
+    @property
+    def arm_joint_pos(self) -> np.ndarray:
+        """Get the current arm joint positions."""
+
+        # deref self vars to cut access in half
+        joint_pos_indices = self.joint_pos_indices
+        arm_joints = self.params.arm_joints
+        sim_obj_joint_pos = self.sim_obj.joint_positions
+
+        arm_pos_indices = (joint_pos_indices[x] for x in arm_joints)
+        return np.array(
+            [sim_obj_joint_pos[i] for i in arm_pos_indices], dtype=np.float32
+        )
+
+    @arm_joint_pos.setter
+    def arm_joint_pos(self, ctrl: List[float]):
+        """Kinematically sets the arm joints and sets the motors to target."""
+        self._validate_arm_ctrl_input(ctrl)
+
+        joint_positions = self.sim_obj.joint_positions
+
+        for i, jidx in enumerate(self.params.arm_joints):
+            self._set_motor_pos(jidx, ctrl[i])
+            joint_positions[self.joint_pos_indices[jidx]] = ctrl[i]
+        self.sim_obj.joint_positions = joint_positions
+
+    def _validate_arm_ctrl_input(self, ctrl: List[float]):
+        """
+        Raises an exception if the control input is NaN or does not match the
+        joint dimensions.
+        """
+        if len(ctrl) != len(self.params.arm_joints):
+            raise ValueError(
+                "Control dimension does not match joint dimension"
```

Comment:
```suggestion
                f"Control dimension ({len(ctrl)}) does not match joint dimension ({len(self.params.arm_joints)})"
```

or something along those lines.